pkgs/development/python-modules/setuptools/default.nix

```diff
@@ -22,6 +23,13 @@ buildPythonPackage rec {
   patches = [
     ./tag-date.patch
     ./setuptools-distutils-C++.patch
+    (fetchpatch {
+      # PR: https://github.com/pypa/setuptools/pull/4484
+      # Fix error, ModuleNotFoundError: No module named 'distutils', for function byte_compile
+      name = "fix-byte-complie.patch";
+      url = "https://patch-diff.githubusercontent.com/raw/pypa/setuptools/pull/4484.patch";
```

Comment:
It is not clear whether upstream will accept the patch as is, because there's a clear separation between what's in `setuptools/_distutils` (which does not depend on `setuptools`, and whose patches should be sent to https://github.com/pypa/distutils) and the rest. I'd suggest putting the patch in the Nixpkgs tree.